### PR TITLE
[Feature][Connector-V2] Add base source connector code for connector-file-base

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

The base of file source connector

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/BaseFileSourceReader.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            return;
+        }
+        sourceSplits.forEach(source -> {
+            try {
+                readStrategy.read(source.splitId(), output);
+            } catch (Exception e) {
+                throw new RuntimeException("File source read error", e);
+            }
+        });
+        context.signalNoMoreElement();
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/BaseFileSourceReader.java:70]: We should consider that the `File Source Connector` should also support streaming jobs? In the feature, we plan to support read local file in streaming model which is like `Flume`. Of course, you can improve it in the future PR.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+public abstract class AbstractReadStrategy implements ReadStrategy {
+    HadoopConf hadoopConf;
+
+    @Override
+    public void init(HadoopConf conf) {
+        this.hadoopConf = conf;
+    }
+
+    @Override
+    public Configuration getConfiguration(HadoopConf hadoopConf) {
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:44]: I guess the `S3 File Source` and `OSS File Source` will extend this class too. Does the `Configuration` can use for read `S3 File` and `OSS File`?

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:44]: > I guess the `S3 File Source` and `OSS File Source` will extend this class too. Does the `Configuration` can use for read `S3 File` and `OSS File`?

As the current orc and parquet are based on hadoop system products, the official api only provides a read interface based on hadoop file system, if we want to interface with oss or s3, currently only supports reading text files, as far as I know, oss and s3 can also support the hdfs protocol file interface, so I think this method will be used in the future.

For the moment, if you want to read orc or parquet files on top of oss or s3, I can think of two options.
1. download the oss or s3 file locally and read it through the local interface, which is not an elegant solution, but a fast and effective one
2. encapsulate the file stream returned by s3 or oss based on the api of parquet or orc, and disguise it as a hadoop file, which is more elegant, but needs time for research and verification

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/BaseFileSourceReader.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/BaseFileSourceReader.java:70]: I think the preview version only supports offline first, and then continue to upgrade to support streaming model.

File: seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml [hunk 3]
```diff
@@ -73,5 +75,12 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.flink</groupId>
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml:80]: If we use flink hadoop jar, the job will failed in spark. So we should set the `scope` of `hadoop` to `provided`.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml:80]: Yes, you are right, I will change it.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml:80]: Resolved, please review.

**Comment by EricJoy2048** [comment]
+1 LGTM
cc @CalvinKirs

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/ReadStrategyFactory.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    public static ReadStrategy of(String fileType) {
+        try {
+            FileFormat fileFormat = FileFormat.valueOf(fileType.toUpperCase());
+            return fileFormat.getReadStrategy();
+        } catch (IllegalArgumentException e) {
+            log.warn("File source connector not support this file type [{}], it will be treated as a plain text file", fileType);
+            return new TextReadStrategy();
+        }
+    }
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/ReadStrategyFactory.java]: When the user has no configuration we use the default txt, when the user has configuration but we do not support it should throw an error

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/ReadStrategyFactory.java]: Resolved, pls review.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**